assets/javascripts/modules/toggleDynamicFormFields.js

```diff
@@ -1,33 +1,39 @@
 require('jquery');
 
-module.exports = function() {
+module.exports = function () {
   var $dynamicForm = $('form[data-dynamic-form]');
-
-  $dynamicForm.find('*[data-dynamic-fields-hide="true"]').each(function() {
-    if ($(this).prop('checked')) {
-      var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
-      $toggledField.addClass('js-hidden');
-    }
-  });
-
-  $dynamicForm.on('click', '*[data-dynamic-fields]', function() {
-    var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
-    if ($(this).data('dynamic-fields-hide')) {
-      if ($(this).attr('type') === 'checkbox') {
-        $(this).data('dynamic-fields-hide', false);
+   $dynamicForm.find('*[data-dynamic-fields-hide="true"]').each(function () {
+      if ($(this).prop('checked')) {
+          var $toggledField = $('.data-' + $(this).data('dynamic-fields'));
+          $toggledField.addClass('js-hidden');
       }
-
-      $toggledField.addClass('js-hidden');
-      $toggledField.find(':text').val('');
-      $toggledField.find(':checked').prop('checked', false);
-      $toggledField.find('*[data-default]').prop('checked', true);
-    } else {
-      if ($(this).attr('type') === 'checkbox') {
-        $(this).data('dynamic-fields-hide', true);
+      $('.data-' + $(this).data('nested-children')).each(function () {
+          if (!$(this).children('div').hasClass("js-hidden")) {
+              $(this).children('label').children('input[type=checkbox]').data('dynamic-fields-hide', true);
+          }
+      });
+  });
```

Comment:
Is this actually needed here? having a look over other projects and doing some reverse engineering it looks like you only need the data attribute `data-dynamic-fields-hide="true"` or `data-dynamic-fields-hide="false"` on the trigger elements in the markup not on the children. If you remove these attributes from your children then it looks like it will work as intended.